engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -0,0 +1,769 @@
+                String mount = String.format(mountCommand, host + ":" + mountPath, TEMPORARY_SECONDARY_STORE);
+                Script.runSimpleBashScript(mount);
+            }
+        } catch (Exception e) {
+            String msg = "NFS Store URL is not in the correct format";
+            LOGGER.error(msg);
+            throw new CloudRuntimeException(msg);
+
+        }
+    }
+
+    private static List<String> fetchAllHypervisors(Connection conn, Long zoneId) {
+        List<String> hypervisorList = new ArrayList<>();
+        try {
+            PreparedStatement pstmt = conn.prepareStatement(FETCH_DISTINCT_HYPERVISORS_IN_ZONE);
+            if(pstmt != null) {
+                pstmt.setLong(1, zoneId);
+                ResultSet resultSet = pstmt.executeQuery();
+                while (resultSet.next()) {
+                    hypervisorList.add(resultSet.getString(1));
+                }
+            }
+        } catch (SQLException e) {
+            String errMsg = String.format("Failed to fetch distinct hypervisors in zone: %s for SystemVM template registration", zoneId);
+            LOGGER.error(errMsg + String.format("due to: %s", e.getMessage()));
+            throw new CloudRuntimeException(errMsg);
+        }
+        return hypervisorList;
+    }
+
+    private static Long createTemplateObjectInDB(Connection conn, SystemVMTemplateDetails details) {
+        Long id = null;
+        try {
+            PreparedStatement pstmt = conn.prepareStatement(INSERT_VM_TEMPLATE_TABLE);
+            if (pstmt != null) {
+                int i = 1;
+                pstmt.setString(i++, details.getUuid());
+                pstmt.setString(i++, details.getUuid());
+                pstmt.setString(i++, details.getName());
+                pstmt.setDate(i++, details.getCreated());
+                pstmt.setString(i++, details.getUrl());
+                pstmt.setString(i++, details.getChecksum());
+                pstmt.setString(i++, details.getName());
+                pstmt.setString(i++, details.getFormat().toString());
+                pstmt.setLong(i++, details.getGuestOsId());
+                pstmt.setString(i++, details.getHypervisorType().toString());
+                pstmt.setBoolean(i++, details.getHypervisorType() == Hypervisor.HypervisorType.VMware);
+                pstmt.executeUpdate();
+
+                pstmt = conn.prepareStatement("SELECT id FROM vm_template ORDER BY id DESC LIMIT 1");
+                try (ResultSet rs = pstmt.executeQuery()) {
+                    if (rs.next()) {
+                        id = rs.getLong(1);
+                    }
+                } catch (final SQLException e) {
+                    String errMsg = String.format("Failed to fetch template id %s", e.getMessage());
+                    LOGGER.error(errMsg);
+                    throw new CloudRuntimeException(errMsg);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Do we need a `Catch them'all` here?

Reply:
I thought it seemed appropriate, do you see any adverse effect with this approach?